partiql-parser/src/lalr/mod.rs

```diff
@@ -29,31 +29,74 @@ mod grammar {
 
 mod lexer;
 
-use crate::location::ByteOffset;
-use crate::result::ParserError;
+use crate::location::{ByteOffset, BytePosition, LineAndCharPosition};
+use crate::result::{ParserError, UnexpectedToken, UnexpectedTokenData};
 pub use lexer::LexError;
 pub use lexer::LineOffsetTracker;
 pub use lexer::Spanned;
 pub use lexer::Token;
 
-type LexerError = (ByteOffset, lexer::LexError, ByteOffset);
-type LalrpopError<'input> = ParseError<ByteOffset, lexer::Token<'input>, LexerError>;
-pub type ParserResult<'input> = Result<Box<ast::Expr>, ParserError<'input>>;
+type LalrpopError<'input> =
+    ParseError<ByteOffset, lexer::Token<'input>, ParserError<'input, BytePosition>>;
+type LalrpopResult<'input> = Result<Box<ast::Expr>, LalrpopError<'input>>;
+type LalrpopErrorRecovery<'input> =
```

Comment:
Oh interesting. Didn't know lalrpop had some notion of [error recovery](http://lalrpop.github.io/lalrpop/tutorial/008_error_recovery.html). I'm curious as to what other parse errors we could make recoverable.

Reply:
Yeah. This PR add it to `TableReference` (essentially for each thing that can be a "FROM source").

I think we should likely explore other such failure points for each possible clause of SFW, and liley for ExprTerm as well.

It'll be nice to be able to catch multiple parse errors per parse.

Reply:
See #88